beacon-chain/rpc/core/validator.go

```diff
@@ -483,3 +484,122 @@ func (s *Service) SubmitSyncMessage(ctx context.Context, msg *ethpb.SyncCommitte
 	// Wait for p2p broadcast to complete and return the first error (if any)
 	return errs.Wait()
 }
+
+// RegisterSyncSubnetCurrentPeriod registers a persistent subnet for the current sync committee period.
+func RegisterSyncSubnetCurrentPeriod(s beaconState.BeaconState, epoch primitives.Epoch, pubKey []byte, status validator.ValidatorStatus) error {
```

Comment:
Most of these added functions don't appear to have any unit tests.

Reply:
The tests in `validator_test.go` already existed in v1alpha1. We don't add any new tests to the core service because the code is tested indirectly through handlers and v1alpha1 endpoints.